plugins/modules/dw_virtual_warehouse.py

```diff
@@ -0,0 +1,433 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# Copyright 2021 Cloudera, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cloudera.cloud.plugins.module_utils.cdp_common import CdpModule
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: dw_virtual_warehouse
+short_description: Create CDP Data Warehouse Virtual Warehouse
+description:
+    - Create CDP Virtual Warehouse
+author:
+  - "Webster Mudge (@wmudge)"
+  - "Dan Chaffelson (@chaffelson)"
+  - "Saravanan Raju (@raju-saravanan)"
+requirements:
+  - cdpy
+options:
+  id:
+    description:
+      - If an ID is provided, that Virtual Warehouse will be deleted if C(state=absent)
+    type: str
+    required: When state is absent
+  cluster_id:
+    description: ID of cluster where Virtual Warehouse should be created.
+    type: str
+    required: True
+  dbc_id:
+    description: ID of Database Catalog that the Virtual Warehouse should be attached to.
+    type: str
+    required: True
+  type:
+    description: Type of Virtual Warehouse to be created.
+    type: str
+    required: True
+  name:
+    description: Name of the Virtual Warehouse.
+    type: str
+    required: True
+  template:
+    description: Name of configuration template to use.
+    type: str
+    required: False
+  autoscaling_min_nodes:
+    description: Minimum number of available nodes for Virtual Warehouse autoscaling.
+    type: int
+    required: False
+  autoscaling_max_nodes:
+    description: Maximum number of available nodes for Virtual Warehouse autoscaling.
+    type: int
+    required: False
+  common_configs: 
+    description: Configurations that are applied to every application in the service.
+    type: dict
+    required: False
+    contains:
```

Comment:
Apparently, `contains` has now been replaced by `suboptions` -- for example: https://github.com/ansible-collections/azure/blob/dev/plugins/modules/azure_rm_securitygroup.py#L49-L52

Full docs are here: https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#documentation-block